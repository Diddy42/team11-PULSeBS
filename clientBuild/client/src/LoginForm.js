import React from 'react';
import {Redirect} from 'react-router-dom';
import API from './API';

class LoginForm extends React.Component {

    constructor(props) {
        super(props);
        // prefilled user pass fields just for convenience for testing purposes
        this.state = { loginSuccess: false, username: 'test', password: 'test' };
    }
    updateField = (name, value) => {
        this.setState({ [name]: value });
    }

    doLoginCall = (username, password) => {
        API.userLogin(username, password).then( (userObj) => {
            if(userObj !== 0) {
                this.props.setLoggedIn(userObj);  // Mettendo setLoggedIn dopo i setState non visualizza /student
                this.setState({loginSuccess: true});       // need to redirect in render
                this.setState({userRole: userObj.role})
            } else {
                this.setState({loginSuccess: false});
                alert("Wrong username or password");
            }
        }).catch(() => {
            console.log("Fail in user login API");
        }
        );
    }

    doLogin = (event) => {
        event.preventDefault();
        if (this.form.checkValidity()) {
            this.doLoginCall(this.state.username, this.state.password);
        } else {
            this.form.reportValidity();
        }
    }
    
    validateForm = (event) => {
        event.preventDefault();
    }

    render() {
        if (this.state.loginSuccess === true && this.state.userRole === "student") {
            console.log("Studente loggato");
            return <Redirect to='/student'/>;
        } else if (this.state.loginSuccess === true && this.state.userRole === "teacher") {
            return <Redirect to='/teacher'/>;
        } else {
            return <div>
                <form className='form' method={'POST'} data-testid="login-form"
                    onSubmit={this.validateForm} ref={form => this.form = form}>
                    <div className={'form-row'}>
                        <div className={'form-group'}>
                            <label htmlFor='username'>Username</label>
                            <input id='username' className={'form-control'} type='text' required={true}
                                name='username'
                                placeholder="Username"
                                value={this.state.username}
                                onChange={(ev) => this.updateField(ev.target.name, ev.target.value)}
                            />
                        </div>
                        &nbsp;
                        <div className={'form-group'}>
                            <label htmlFor='password'>Password</label>
                            <input id='password' className={'form-control'} type='password' required={true}
                                name='password'
                                value={this.state.password}
                                placeholder="Password"
                                onChange={(ev) => this.updateField(ev.target.name, ev.target.value)}
                            />
                        </div>
                    </div>
                    <div className={'form-row'}>
                        <button type='button' className='btn btn-primary' disabled={this.props.doingLogin}
                            onClick={this.doLogin}>Login</button>
                    </div>
                </form>
            </div>
        }
}

}

export default LoginForm