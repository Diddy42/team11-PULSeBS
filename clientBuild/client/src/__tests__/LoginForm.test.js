import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import LoginForm from '../LoginForm';

test("LoginFrom renders correctly", () => {
    const {queryByTestId} = render(<LoginForm/>);
    expect(queryByTestId("login-form")).toBeTruthy();
});

describe("Input value", () => {
    test("update on change username", () => {
        const {queryByPlaceholderText} = render(<LoginForm/>);
        const usernameInput = queryByPlaceholderText("Username");
        const passwordInput = queryByPlaceholderText("Password");
        fireEvent.change(usernameInput, {target: {value: "calogero"}});
        expect(usernameInput.value).toBe("calogero");
        fireEvent.change(passwordInput, {target: {value: "test"}});
        expect(passwordInput.value).toBe("test");
    });
});