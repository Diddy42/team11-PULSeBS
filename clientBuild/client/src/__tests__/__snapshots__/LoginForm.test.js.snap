// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFrom renders correctly 1`] = `
<div>
  <form
    className="form"
    data-testid="login-form"
    method="POST"
    onSubmit={[Function]}
  >
    <div
      className="form-row"
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="username"
        >
          Username
        </label>
        <input
          className="form-control"
          id="username"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
          value="test"
        />
      </div>
      Â 
      <div
        className="form-group"
      >
        <label
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value="test"
        />
      </div>
    </div>
    <div
      className="form-row"
    >
      <button
        className="btn btn-primary"
        data-testid="login-button"
        onClick={[Function]}
        type="button"
      >
        Login
      </button>
    </div>
  </form>
</div>
`;
